### ISSUE TITLE
Problem with date format

### ISSUE THREAD

**Issue by robmcfadden** [barrelstrength/craft-sprout-import#58]
### Description

I'm getting `Call to a member function getTimestamp() on string`. I've tried multiple date formats for `postDate` and `expiryDate`, but haven't had any luck. Is there a bug here? Or do I need to use a specific date format? Thanks!

### Steps to reproduce

1. Import an entry with a `postDate` or `expiryDate` included.
2. Get this error: `Call to a member function getTimestamp() on string`

### Additional info

- Craft version: 3.0.18
- PHP version: 7.1.16
- Database driver & version: MySQL 5.7.22
- Plugins & versions: 1.0.0-beta.14

**Comment by BenParizek** [comment]
It appears `getTimestamp` is being triggered by Craft's `getStatus` method:

``` 2018-08-27 10:06:42 [-][-][-][error][craft\queue\QueueLogBehavior::afterError]  [862] Importing Data. (attempt: 1) - Error (time: 0.859s): Call to a member function getTimestamp() on string
2018-08-27 10:06:42 [-][-][-][error][Error] Error: Call to a member function getTimestamp() on string in /app/vendor/composer/craftcms/cms/src/elements/Entry.php:760
Stack trace:
#0 /app/vendor/composer/yiisoft/yii2/base/Component.php(139): craft\elements\Entry->getStatus()
#1 /app/vendor/composer/craftcms/cms/src/base/Element.php(797): yii\base\Component->__get('status')
#2 /app/vendor/composer/yiisoft/yii2/base/ArrayableTrait.php(126): craft\base\Element->__get('status')```

so yeah - getStatus does this: `$postDate = $this->postDate->getTimestamp();`
so `postDate` must still be a string at  that point and not DateTime like it should be

**Comment by timkelty** [comment]
@BenParizek here's the full trace:

```
2018-08-27 10:06:42 [-][-][-][error][craft\queue\QueueLogBehavior::afterError]  [862] Importing Data. (attempt: 1) - Error (time: 0.859s): Call to a member function getTimestamp() on string
2018-08-27 10:06:42 [-][-][-][error][Error] Error: Call to a member function getTimestamp() on string in /app/vendor/composer/craftcms/cms/src/elements/Entry.php:760
Stack trace:
#0 /app/vendor/composer/yiisoft/yii2/base/Component.php(139): craft\elements\Entry->getStatus()
#1 /app/vendor/composer/craftcms/cms/src/base/Element.php(797): yii\base\Component->__get('status')
#2 /app/vendor/composer/yiisoft/yii2/base/ArrayableTrait.php(126): craft\base\Element->__get('status')
#3 /app/vendor/composer/craftcms/cms/src/web/View.php(496): yii\base\Model->toArray(Array, Array, false)
#4 /app/vendor/composer/craftcms/cms/src/helpers/ElementHelper.php(149): craft\web\View->renderObjectTemplate('events/{{ (_var...', Object(craft\elements\Entry), Array)
#5 /app/vendor/composer/craftcms/cms/src/helpers/ElementHelper.php(92): craft\helpers\ElementHelper::_renderUriFormat('events/{slug}', Object(craft\elements\Entry))
#6 /app/vendor/composer/craftcms/cms/src/validators/ElementUriValidator.php(49): craft\helpers\ElementHelper::setUniqueUri(Object(craft\elements\Entry))
#7 /app/vendor/composer/yiisoft/yii2/validators/Validator.php(267): craft\validators\ElementUriValidator->validateAttribute(Object(craft\elements\Entry), 'uri')
#8 /app/vendor/composer/yiisoft/yii2/base/Model.php(367): yii\validators\Validator->validateAttributes(Object(craft\elements\Entry), Array)
#9 /app/vendor/composer/barrelstrength/sprout-import/src/services/ElementImporter.php(132): yii\base\Model->validate(Array, false)
#10 /app/vendor/composer/barrelstrength/sprout-base/src/app/import/services/Importers.php(308): barrelstrength\sproutimport\services\ElementImporter->saveElement(Array, Object(barrelstrength\sproutimport\importers\elements\Entry))
#11 /app/vendor/composer/barrelstrength/sprout-import/src/queue/jobs/Import.php(43): barrelstrength\sproutbase\app\import\services\Importers->save(Array, Object(barrelstrength\sproutimport\models\Weed))
#12 /app/vendor/composer/yiisoft/yii2-queue/src/Queue.php(214): barrelstrength\sproutimport\queue\jobs\Import->execute(Object(craft\queue\Queue))
#13 /app/vendor/composer/yiisoft/yii2-queue/src/cli/Queue.php(162): yii\queue\Queue->handleMessage('862', 'O:45:"barrelstr...', '300', '1')
#14 /app/vendor/composer/yiisoft/yii2-queue/src/cli/Command.php(145): yii\queue\cli\Queue->execute('862', 'O:45:"barrelstr...', '300', '1', '')
#15 [internal function]: yii\queue\cli\Command->actionExec('862', '300', '1', '')
#16 /app/vendor/composer/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#17 /app/vendor/composer/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#18 /app/vendor/composer/yiisoft/yii2/console/Controller.php(148): yii\base\Controller->runAction('exec', Array)
#19 /app/vendor/composer/yiisoft/yii2/base/Module.php(528): yii\console\Controller->runAction('exec', Array)
#20 /app/vendor/composer/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('queue/exec', Array)
#21 /app/vendor/composer/yiisoft/yii2/console/Application.php(147): yii\console\Application->runAction('queue/exec', Array)
#22 /app/vendor/composer/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest(Object(craft\console\Request))
#23 /app/vendor/composer/fusionary/craftcms-bootstrap/src/Bootstrap.php(76): yii\base\Application->run()
#24 /app/bin/craft(4): fusionary\craftcms\bootstrap\Bootstrap::run('console')
#25 {main}
```

**Comment by pdaleramirez** [comment]
@BenParizek this is a bug on craft 3 core code. When you add the expiryDate value it throws an error. This works in craft 2 sprout import. We can submit this issue.

**Comment by timkelty** [comment]
> When you add the expiryDate value it throws an error

Separate from Sprout Import? I can add an expiry date to an entry just fine…

**Comment by pdaleramirez** [comment]
I found out that the issue is on sprout-import and not on craft. I added the fix here https://github.com/barrelstrength/craft-sprout-base/commit/8de6c6de88dda1ff7fd153674a1070808a247046

**Comment by BenParizek** [comment]
@robmcfadden This is resolved in v1.0.0-beta.15, now available via the Plugin Store
